src/hotspot/share/opto/node.cpp

```diff
@@ -1601,6 +1602,199 @@ Node* old_root() {
   return nullptr;
 }
 
+// UniqueMixedNodeList
+// unique: nodes are added only once
+// mixed: allow new and old nodes
+class UniqueMixedNodeList {
+public:
+  UniqueMixedNodeList() : _visited_set(cmpkey, hashkey) {}
+  void add(Node* node) {
+    if (not_a_node(node)) {
+      return; // Gracefully handle NULL, -1, 0xabababab, etc.
+    }
+    if (_visited_set[node] == nullptr) {
+      _visited_set.Insert(node, node);
+      _worklist.push(node);
+    }
+  }
+  Node* operator[] (uint i) const {
+    return _worklist[i];
+  }
+  size_t size() {
```

Comment:
You should add some new lines to better separate the methods.

Reply:
done, thanks